sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuShuffleEnv.scala

```diff
@@ -104,11 +104,22 @@ object GpuShuffleEnv extends Logging {
   // this forces the initialization when we know we are ready in the driver and executor.
   //
   def initShuffleManager(): Unit = {
-    SparkEnv.get.shuffleManager match {
+    val shuffleManager = SparkEnv.get.shuffleManager
+    shuffleManager match {
       case rapidsShuffleManager: RapidsShuffleManagerLike =>
         rapidsShuffleManager.initialize
       case _ =>
-        throw new IllegalStateException(s"Cannot initialize the RAPIDS Shuffle Manager")
+        val rsmLoaderViaShuffleManager = shuffleManager.getClass.getSuperclass.getInterfaces
+          .collectFirst {
+            case c if c.getName == classOf[RapidsShuffleManagerLike].getName => c.getClassLoader
+          }
+        val rsmLoaderDirect = classOf[RapidsShuffleManagerLike].getClassLoader
+ 
+        throw new IllegalStateException(s"Cannot initialize the RAPIDS Shuffle Manager " +
+          s"${shuffleManager}! Expected: an instance of  RapidsShuffleManagerLike loaded by " +
+          s"${rsmLoaderDirect}. Actual: ${shuffleManager} tagged with RapidsShuffleManagerLike " +
+          s"loaded by: ${rsmLoaderViaShuffleManager}"
```

Comment:
if the shuffle manager loaded isn't a `RapidsShuffleManagerLike` then I think this is kind of confusing. I am not 100% sure we can have this case. I believe it would read: "Actual: com.nvidia.spark.rapids.sparkXYZ.RapidsShuffleManager tagged with RapidsShuffleManagerLike loaded by: None".

If that's the case, I suggest we check whether `rsmLoaderViaShuffleManager` is defined and adjust the message accordingly. If we remove the "tagged with RapidsShuffleManagerLike" that would be good if we don't find it in the interfaces.

Reply:
I don't think that is possible because if it is not a `RapidsShuffleManagerLike`, then `RapidsShuffleManagerLike` will return false when called from `RapidsExecutorPlugin` [here](https://github.com/NVIDIA/spark-rapids/blob/d1e986d6b7ab5cdb44b68a601f566c17ce3c9bcf/sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala#L462) and thus `initShuffleManager` would not get called.

Reply:
sounds good